### PR TITLE
Clean up unused sub-table

### PR DESCRIPTION
**Pull request by beeender**
sub-table is not used in java-binding right now, and if it is going to be, the relevant logic might be more suitable in the object store.

Since we are moving to the object store, the sub-table related code needs to be cleaned anyway. So we do the clean up directly to the master branch to make the future PR easier to be reviewed.

### CODE REVIEW

File: realm/realm-jni/src/io_realm_internal_TableView.cpp [hunk 1]
```diff
@@ -1035,8 +999,7 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableView_nativeWhere(
         if (!VIEW_VALID_AND_IN_SYNC(env, nativeViewPtr))
             return 0;
 
-        Query query = TV(nativeViewPtr)->get_parent().where(TV(nativeViewPtr));
-        TableQuery* queryPtr = new TableQuery(query);
+        Query *queryPtr = new Query(std::move(TV(nativeViewPtr)->get_parent().where(TV(nativeViewPtr))));
```
Comment (cmelchior) [inline realm/realm-jni/src/io_realm_internal_TableView.cpp]: Any reason for changing this from a TableQuery to Query?

File: realm/realm-jni/src/io_realm_internal_TableQuery.cpp [hunk 2]
```diff
@@ -38,7 +37,7 @@ inline bool query_valid(JNIEnv* env, Query* pQuery)
 
 inline bool query_col_type_valid(JNIEnv* env, jlong nativeQueryPtr, jlong colIndex, DataType type)
 {
-    return TBL_AND_COL_INDEX_AND_TYPE_VALID(env, TQ(nativeQueryPtr)->get_current_table().get(), colIndex, type);
+    return TBL_AND_COL_INDEX_AND_TYPE_VALID(env, Q(nativeQueryPtr)->get_table().get(), colIndex, type);
```
Comment (cmelchior) [inline realm/realm-jni/src/io_realm_internal_TableQuery.cpp]: Same, why this change?

File: realm/realm-jni/src/io_realm_internal_TableView.cpp (hunk 1 above)
Comment (beeender) [inline realm/realm-jni/src/io_realm_internal_TableView.cpp]: `TableQuery` is created only for JNI and to wrap around the subtable. And it is deleted in this PR as well.

**Comment by cmelchior** [comment]
Looks 👍 
But why the change from TableQuery to Query?

File: realm/realm-jni/src/io_realm_internal_TableView.cpp (hunk 1 above)
Comment (cmelchior) [inline realm/realm-jni/src/io_realm_internal_TableView.cpp]: 👍

File: realm/realm-jni/src/io_realm_internal_TableQuery.cpp (hunk 2 above)
Comment (stk1m1) [inline realm/realm-jni/src/io_realm_internal_TableQuery.cpp]: I think it is b/c TableQuery is to be removed. In TQ, we have

```c++
realm::TableRef get_current_table() {
    realm::TableRef table = get_table();

    // Go through the stack of subtables to find current subtable (if any)
    size_t size = subtables.size(); 
    for (size_t i = 0; i < size; ++i) {
        size_t index = subtables[i];
        table = table->get_subtable(index, 0);
    }
    return table;
}
```

<sup>*</sup>There is _finding current table_ logic

Comment (kneth) [inline realm/realm-jni/src/io_realm_internal_TableQuery.cpp]: Yes, when you query a subtable, you will need to get a reference to it. `get_current_table()` descents down to the subtable as the query only has access to the parent table.

File: realm/realm-jni/src/io_realm_internal_TableView.cpp (hunk 1 above)
Comment (kneth) [inline realm/realm-jni/src/io_realm_internal_TableView.cpp]: Well, `TableQuery` was also introduced so we could do a deep copy of the query. @beeender are you certain that the core's `Query` constructor does a deep copy?

**Comment by kneth** [comment]
@cmelchior `TableQuery` was partly introduced to keep track of subtable queries (balancing begin and end markers).

File: realm/realm-jni/src/io_realm_internal_TableView.cpp (hunk 1 above)
Comment (beeender) [inline realm/realm-jni/src/io_realm_internal_TableView.cpp]: It is calling move constructor and confirmed with core, it should work :tada:

**Comment by stk1m1** [comment]
Other than what @kneth pointed, 👍

**Comment by bmunkholm** [comment]
@beeender Please include motivation in the description besides just the title.
It's not clear that it has been confirmed that we are not going to need subtables for any of the other future datatypes we want to support.

**Comment by beeender** [comment]
@bmunkholm PR message updated.

File: realm/realm-jni/src/io_realm_internal_LinkView.cpp [hunk 3]
```diff
@@ -178,8 +177,7 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_LinkView_nativeWhere
     try {
         LinkViewRef *lv = LV(nativeLinkViewPtr);
         LinkViewRef lvr = *lv;
-        Query query = lvr->get_target_table().where(LinkViewRef(lvr));
-        TableQuery* queryPtr = new TableQuery(query);
+        Query *queryPtr = new Query(std::move(lvr->get_target_table().where(LinkViewRef(lvr))));
```
Comment (bdash) [inline realm/realm-jni/src/io_realm_internal_LinkView.cpp]: The use of `std::move` here is unnecessary since `lvr->get_target_table().where(LinkViewRef(lvd))` is already an rvalue.

File: realm/realm-jni/src/io_realm_internal_Table.cpp [hunk 4]
```diff
@@ -1159,8 +1065,7 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_Table_nativeWhere(
     if (!TABLE_VALID(env, TBL(nativeTablePtr)))
         return 0;
     try {
-        Query query = TBL(nativeTablePtr)->where();
-        TableQuery* queryPtr = new TableQuery(query);
+        Query *queryPtr = new Query(std::move(TBL(nativeTablePtr)->where()));
```
Comment (bdash) [inline realm/realm-jni/src/io_realm_internal_Table.cpp]: Ditto about `std::move` being unnecessary.
